### PR TITLE
Add so.libdb.dissent

### PR DESCRIPTION
**Pull request by tfuxu**
### Please confirm your submission meets all the criteria

Please describe your application briefly below.

Dissent is a new name for [Gtkcord4](https://github.com/flathub/so.libdb.gtkcord4), unofficial Discord client written with Gtk4 and Go.

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: so.libdb.dissent.yml [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+      # Workaround for Go modules generated by github.com/dennwc/flatpak-go-mod
+      - type: file
+        path: modules.txt
+        dest: vendor
+
+      - type: archive
+        url: https://proxy.golang.org/github.com/diamondburned/adaptive/@v/v0.0.2-0.20221227093656-fa139be203a8.zip
+        strip-components: 3
+        dest: vendor/github.com/diamondburned/adaptive
```
Comment (bbhtt) [inline so.libdb.dissent.yml:54]: These should be separated from the main manifest, you are auto generating them.

Comment (tfuxu) [reply so.libdb.dissent.yml:54]: How can I do that?

Comment (bbhtt) [reply so.libdb.dissent.yml:54]: See https://github.com/flathub/io.github.dweymouth.supersonic/blob/master/go.mod.yml or https://github.com/flathub/com.visualstudio.code.tool.git-lfs/blob/branch/23.08/vendor-sources.json

**Comment by tfuxu** [comment]
bot, build

**Comment by tfuxu** [comment]
bot, build so.libdb.dissent

**Comment by diamondburned** [comment]
Sorry for bothering, but is there any update on this?

File: so.libdb.dissent.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  - org.freedesktop.Sdk.Extension.golang
+
+finish-args:
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --socket=pulseaudio
+  - --share=ipc
+  - --share=network
+  - --talk-name=org.freedesktop.Notifications
+  - --talk-name=org.freedesktop.secrets
```
Comment (bbhtt) [inline so.libdb.dissent.yml:17]: Are these needed? Or do they work through the portal?

File: so.libdb.dissent.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  - --talk-name=org.kde.kwalletd5
+  - --talk-name=org.kde.StatusNotifierWatcher
+  - --filesystem=xdg-run/keyring  # TODO: Check if this is even needed
```
Comment (bbhtt) [inline so.libdb.dissent.yml:20]: Same as above

**Comment by tfuxu** [comment]
@diamondburned How do we store user credentials in Dissent?

**Comment by diamondburned** [comment]
It currently uses [zalando/go-keyring](https://github.com/zalando/go-keyring) which uses its own DBus connection independent from the main GTK application. I mostly couldn't figure out how to adapt go-keyring to GTK.

**Comment by diamondburned** [comment]
https://github.com/diamondburned/chatkit/blob/4b8acdf0f84b/kits/secret/keyring.go#L24 says `so.libdb.dissent.secrets` is the DBus name.

**Comment by bbhtt** [comment]
go-keyring doesn't seem to use the secret portal, my concern was if it did.

**Comment by bbhtt** [comment]
bot, build

**Comment by bbhtt** [comment]
bot, build so.libdb.dissent

**Comment by bbhtt** [comment]
Seems fine from my side.

**Pull request closed without merging**